sympy/integrals/tests/test_manual.py

```diff
@@ -191,11 +191,17 @@ def test_manualintegrate_inversetrig():
     # asin
     assert manualintegrate(asin(x), x) == x*asin(x) + sqrt(1 - x**2)
     assert manualintegrate(asin(a*x), x) == Piecewise(((a*x*asin(a*x) + sqrt(-a**2*x**2 + 1))/a, Ne(a, 0)), (0, True))
-    assert manualintegrate(x*asin(a*x), x) == -a*Integral(x**2/sqrt(-a**2*x**2 + 1), x)/2 + x**2*asin(a*x)/2
+    assert_is_integral_of(x*asin(a*x),
+                          -a*(-x*sqrt(-a**2*x**2 + 1)/(2*a**2) +
+                              log(-2*a**2*x + 2*sqrt(-a**2)*sqrt(-a**2*x**2 + 1))/(2*a**2*sqrt(-a**2)))/2 +
+                          x**2*asin(a*x)/2)
     # acos
     assert manualintegrate(acos(x), x) == x*acos(x) - sqrt(1 - x**2)
     assert manualintegrate(acos(a*x), x) == Piecewise(((a*x*acos(a*x) - sqrt(-a**2*x**2 + 1))/a, Ne(a, 0)), (pi*x/2, True))
-    assert manualintegrate(x*acos(a*x), x) == a*Integral(x**2/sqrt(-a**2*x**2 + 1), x)/2 + x**2*acos(a*x)/2
+    assert_is_integral_of(x*acos(a*x),
+                          a*(-x*sqrt(-a**2*x**2 + 1)/(2*a**2) +
```

Comment:
Shouldn't this be a Piecewise like:
```python
In [10]: integrate(x*acos(a*x), x)
Out[10]: 
⎧                    _____________                       
⎪ 2                 ╱    2  2                            
⎪x ⋅acos(a⋅x)   x⋅╲╱  - a ⋅x  + 1    acos(a⋅x)           
⎪──────────── - ────────────────── - ─────────  for a ≠ 0
⎪     2                4⋅a                 2             
⎨                                       4⋅a              
⎪                                                        
⎪                       2                                
⎪                    π⋅x                                 
⎪                    ────                       otherwise
⎩                     4                                  
```

Reply:
https://github.com/sympy/sympy/issues/23550#issuecomment-1140262403
Do we want to handle degenerated cases?

Reply:
What do you mean by referring to my other comment?

I asked there whether it makes sense to explicitly list the degenerate cases in every rule or whether it would be better to use recursive calls to handle the degenerate cases. That's about the internal implementation and is separate from the question of whether we want to return expressions that can handle degenerate cases.

Reply:
OK, I misunderstood. Then I prefer recursive calls for less complexity.